server/routes/index.js

```diff
@@ -0,0 +1,106 @@
+const fs = require('fs')
+
+const each = require('async/each')
+const pick = require('lodash.pick')
+
+const routes = {}
+module.exports = routes
+
+routes.makeRoutes = (app) => {
+  fs.readdirSync(__dirname).forEach((file) => {
+    if (file === 'index.js') return
+    const name = file.substr(0, file.indexOf('.'))
+    require('./' + name)(app)
+  })
+}
+
+const serverError = (res, err) => {
+  console.error(err)
+  res.status(500).json({error: err})
+}
+
+const makeGenericModelResponse = (res) => (err, data) => {
+  res.set('Content-Type', 'application/json')
+  if (err) return serverError(res, err)
+  res.json(data)
+}
+
+const makeGetModelResponse = (childModels, res, isCollection) => (err, data) => {
+  const genericResponder = makeGenericModelResponse(res)
+  if (!Array.isArray(data)) {
+    data = [data]
+  }
+
+  if (!childModels) return genericResponder(err, isCollection ? data : data[0])
+
+  // wow, awesome, with normalized mongoose models I get to make extra queries
+  // to the db to do joins!  </sarcasm>
```

Comment:
Yes, they were painful to write.  And it'll be slower since it's more roundtrips to the database.  And there's no referential integrity.

If I use `populate` on arrays of children ids then I have to refactor all of the models that have children models.  I'd have to do even more manual work of adding the ids of new models to their parent and then deleting them from their parent's children array upon deletion.  Seems like more work that way.